cell/src/main/java/jetbrains/jetpad/cell/text/TextEditingTrait.java

```diff
@@ -181,16 +181,13 @@ public void onPaste(Cell cell, PasteEvent event) {
 
     String text = content.get(ContentKinds.TEXT);
 
-    StringBuilder newText = new StringBuilder();
-    for (int i = 0; i < text.length(); i++) {
-      if (text.charAt(i) != '\n') {
-        newText.append(text.charAt(i));
-      }
+    if (text.contains("\n") || text.contains("\r")) {
```

Comment:
Me and Alexei agreed that a separate multiline text content kind is a way to go. Closing this pull request.